src/inapp/inapp.ts

```diff
@@ -353,31 +353,24 @@ export function getInAppMessages(
             Also swallow any 400+ response errors. We don't care about them.
           */
           if (ENABLE_INAPP_CONSUME || IS_PRODUCTION) {
-            Promise.all(
-              !activeMessage?.saveToInbox
-                ? [
-                    trackInAppOpen({
-                      messageId: activeMessage.messageId,
-                      deviceInfo: {
-                        appPackageName: payload.packageName
-                      }
-                    }),
-                    trackInAppConsume({
-                      messageId: activeMessage.messageId,
-                      deviceInfo: {
-                        appPackageName: payload.packageName
-                      }
-                    })
-                  ]
-                : [
-                    trackInAppOpen({
-                      messageId: activeMessage.messageId,
-                      deviceInfo: {
-                        appPackageName: payload.packageName
-                      }
-                    })
-                  ]
-            ).catch((e) => e);
+            const trackRequests = [
+              trackInAppOpen({
+                messageId: activeMessage.messageId,
+                deviceInfo: {
+                  appPackageName: payload.packageName
+                }
+              })
+            ];
+            if (!activeMessage.saveToInbox)
+              trackRequests.push(
+                trackInAppConsume({
+                  messageId: activeMessage.messageId,
+                  deviceInfo: {
+                    appPackageName: payload.packageName
+                  }
+                })
+              );
+            Promise.all(trackRequests).catch((e) => e);
```

Comment:
Opting to push trackInAppConsume to the trackRequests array which always starts with trackInAppOpen to reduce duplicate code.